pipeline/common/downloads.py

```diff
@@ -299,10 +299,15 @@ def download_chunks(self) -> Generator[bytes, None, None]:
                 # the retries are done.
                 logger.error(f"A download error occurred: {error}")
 
```

Comment:
This is the codepath that runs when a response times out, or a network connection is lost.

Reply:
I'm not sure if "with" can be applied here to ensure the resources will be closed in any case

Reply:
I have been playing around with `@contextmanager`(I think you pointed that out) and `ExitStack`. This could would probably be simpler with that.